api/scpca_portal/models/job.py

```diff
@@ -308,6 +308,22 @@ def update_state_at(self, save: bool = True) -> None:
         if save:
             self.save()
 
+    def set_state(self, state: JobStates, reason: str = None):
+        """
+        Sets the job state and its corresponding timestamp.
+        Calls the associated dataset's on_job_<state> event handler.
+        """
+        state_str = state.lower()
+        reason_attr = f"{state_str}_reason"
+        event_handler = f"on_job_{state_str}"
+
+        self.state = state
+        if hasattr(self, reason_attr):
+            setattr(self, reason_attr, reason)
+        self.update_state_at()
+
+        getattr(self.dataset, event_handler)()
```

Comment:
Let's comment this out with a TODO and we can discuss sync this sprint about updating the dataset.

Reply:
Looking forward to it!